packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -54,6 +55,16 @@ class ModalBarrier extends StatelessWidget {
   ///    [ModalBarrier] built by [ModalRoute] pages.
   final bool dismissible;
 
+  /// Called when the barrier is being dismissed.
+  ///
+  /// If non-null [onDismiss] will be called in place of popping the current
+  /// route. It is up to the callback to handle dismissing the barrier.
+  ///
+  /// If null, the ambient [Navigator]'s current route will be popped.
+  ///
+  /// This field is ignored if [dismissible] is false.
```

Comment:
I was going to say that maybe we should assert that onDismiss is null when `dismissible` is false, but now I'm thinking that might be more complicated for users if they're toggling `dismissible`...  Probably best to keep it like this.